File: hyejee/week4/위장.swift
import Foundation

func solution(_ clothes:[[String]]) -> Int {
    var option: [String: [String]] = [:]

    // parse clothes array
    for _clothes in clothes {
        option[_clothes[1], default: []].append(_clothes[0])
    }

    // calculate combination
    var answer: Int = 1

    for listOfOptions in option.values {
        // choose one option in listOfOptions or choose none
        answer *= (1 + listOfOptions.count)
    }

    // remove a case that choose nothing
    answer -= 1

    return answer
}

File: chaewon/week4/위장.py
def solution(clothes):
    closet = {}
    for cloth in clothes:
        if cloth[1] in closet:
            closet[cloth[1]] += 1
        else:
            closet[cloth[1]] = 1

    answer = 1
    for i in closet.values():
        answer *= (i + 1)
    return answer - 1